tests/Tests.AzureAppConfiguration/Tests.cs

```diff
@@ -263,5 +264,37 @@ public void TestTurnOffRequestTracing()
             // Delete the azure function environment variable
             Environment.SetEnvironmentVariable(RequestTracingConstants.AzureFunctionEnvironmentVariable, null);
         }
+
+        [Fact]
+        public void DisableProviderWithEnvironmentVariable()
+        {
+            var mockResponse = new Mock<Response>();
+            var mockClient = new Mock<ConfigurationClient>(MockBehavior.Strict);
+
+            mockClient.Setup(c => c.GetConfigurationSettingsAsync(It.IsAny<SettingSelector>(), It.IsAny<CancellationToken>()))
+                .Returns(new MockAsyncPageable(_kvCollectionPageOne));
+
+            Environment.SetEnvironmentVariable(ConditionalProviderConstants.DisableProviderEnvironmentVariable, "true");
```

Comment:
This could lead to other tests failing when they are run in parallel. We need a way to run this test such that it never runs in parallel with any other test. Can you check if there's a way to run this one test sequentially and not in parallel?

Reply:
This answer had some suggestions about using `Collection` attribute, but that may not work for our scenario: https://stackoverflow.com/questions/1408175/execute-unit-tests-serially-rather-than-in-parallel

Reply:
Oh ok thanks, yea I didn't consider that I'll look into it

Reply:
I haven't been able to find anything too helpful on this. I think it might just be a good idea to remove this test for now, since the only solutions I've seen that might work complicate things significantly.

Reply:
Agreed, its not worth taking on the additional complexity.